### DISCUSSION TITLE
Postfix hold queue

### DISCUSSION THREAD

**Discussion by andrewlow** [docker-mailserver/docker-mailserver#3611] [Q&A] [answered]
Can someone educate me on the postfix hold queue?

I just discovered that I had a bunch (35) emails queued up .. spanning back for some time (most of this year). I've been running dockermailserver for a long time, and I didn't see any from way way back.. so this may be a more recent problem -- but one from months ago.

Exec'ing into the container and running `mailq` showed me the 35 queued items.

Running `postsuper -H ALL` caused all of these messages to be delivered after a minute or two.

However, I don't really understand how these messages got stuck in the hold queue. The emails were valid, but some were from a mailing list. 

I started digging into this because I do have logwatch configured  (`LOGWATCH_INTERVAL=daily`) - and saw this

```
--------------------- Postfix Begin ------------------------ 

       2   Placed on hold                                   2
```

which got me investigating..  

However, I'm wondering if I should be paying more close attention to messages put on hold .. and clearing the queue regularly or what.

**Comment by casperklein** [comment, marked as answer]
> However, I don't really understand how these messages got stuck in the hold queue. 

Network issues or unavailable destination mail server can be the reason. After some unsuccessful tries, postfix puts them on hold and don't start further delivery attempts. You'll have to check your mail logs to know, why they were put on hold.

**Reply by andrewlow** [reply]
Ok - well, that happens from time to time (network issues.. ) mostly I need to pay closer attention to any growth of the `mailq` so I can go take care of those things. Maybe I'll create a nightly script to make sure that anything queued.. is re-kicked to get delivered, and I'll get notified.

Interestingly - a bunch of the emails were from a mailing list to me.. odd that those specifically were delayed.. but as you say, I need to see the logs.. but what I need.. is to know when to go looking.

**Reply by casperklein** [reply]
> Maybe I'll create a nightly script to make sure that anything queued.

You can enable report options pflogsumm/logwatch that should contain infos about mails on hold.
